test/common/upstream/subset_lb_test.cc

```diff
@@ -283,7 +283,7 @@ class SubsetLoadBalancerTest : public testing::TestWithParam<UpdateOrder> {
       host_set.healthy_hosts_per_locality_ = host_set.hosts_per_locality_;
     }
 
-    if (GetParam() == REMOVES_FIRST && !remove.empty()) {
+    if (GetParam() == UpdateOrder::RemovesFirst && !remove.empty()) {
```

Comment:
Can we enforce having the enum class' namespace in uses of the enum values?

Reply:
I believe enum classes do require the namespace, but the problem here is that `UpdateOrder` was a plain enum, not an enum class (line 105). Enum classes should be preferred ([CppCoreGuideline Enum.3](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Renum-class)) since it enforces namespacing the values and has stricter typing.

I don't think there's an explicit check to ensure we're using enum classes over plain enums, that's something I could look into on clang-tidy's side.